src/gbm/gblinear.cc

```diff
@@ -317,60 +226,51 @@ class GBLinear : public GradientBooster {
   }
 
  protected:
-  inline void Pred(const RowBatch::Inst &inst, bst_float *preds, int gid, bst_float base) {
+  bool CheckConvergence() {
+    if (is_converged) return true;
```

Comment:
Is_converged || param.tolerance == 0.

Reply:
I think IsConverged should always return false if tol == 0. The updater does not run if IsConverged== true.

Reply:
Yep, my bad...  But shouldn't then the line be `if (param.tolerance == 0.f) return false;`?

Reply:
Yes thanks